### PR TITLE
various bug fixes

### PR DESCRIPTION
**Pull request by M0ses**

### CODE REVIEW

File: 20-files-present-and-referenced [hunk 1]
```diff
@@ -20,7 +20,9 @@ test -x /usr/bin/rpmbuild && RPMBUILD=rpmbuild
 #  cleanup_and_exit
 #
 cleanup_and_exit () {
-    rm -rf $TMPDIR
+    if [[ -n $TMPDIR ]];then
+    	rm -rf $TMPDIR
+    fi
     exit $1
 }
```
Comment (M0ses) [inline 20-files-present-and-referenced]: Rest in this file are only "trailing whitespace fixes

File: helpers/output_versions [hunk 2]
```diff
@@ -9,31 +12,30 @@ my $add_license = 0;
 my $debug = 0;
 my $dir_to_check;
 
-for (@ARGV) {
-	if ($_ eq "-q") {
```
Comment (M0ses) [inline helpers/output_versions]: Changes below are to fix the "-r" option, in original source $release was always "-r"

File: helpers/output_versions [hunk 3]
```diff
@@ -9,31 +12,30 @@ my $add_license = 0;
...
-		$dir_to_check = $_;
+		$dir_to_check = $opt;
 		last;
 	}
 }
 
 sub output_package ($$) {
 	my ($rel,$cpack) = @_;
-	my $datestr = `date "+%Y-%m-%d %H:%M:%S"`;
-	chomp ($datestr);
```
Comment (M0ses) [inline helpers/output_versions]: prevent system call

File: helpers/output_versions [hunk 4]
```diff
@@ -66,10 +64,16 @@ for (readdir(NEWD)) {
 closedir (NEWD);
 
 for my $specfile (sort(@specs)) {
-	my $buildroot = 0;
-	$main_package_name = "";
-        undef $definelist;
-        undef $ifhandler;
```
Comment (M0ses) [inline helpers/output_versions]: declare variables in right scope as "use strict" now in place

File: 20-files-present-and-referenced (hunk 1 above)
Comment (bugfinder) [inline 20-files-present-and-referenced]: please do not use bash 4 syntax, this breaks on too many older systems.
if [ -n "$TMPDIR" ] ; then
...

**Comment by bugfinder** [comment]
please fix the hunk in the bash script, the rest looks perfectly okay.

**Comment by M0ses** [comment]
@bugfinder : removed bash 4 syntax

**Comment by bugfinder** [comment]
ack looks good

**Pull request merged**